### PR TITLE
[DOCS] Reformat `stemmer` token filter

### PR DESCRIPTION
**Pull request by jrodewig**
Makes the following changes to the `stemmer` token filter docs:

* Adds detailed analyze example
* Rewrites parameter definitions
* Adds custom analyzer example
* Adds a `language` value for the `estonian` stemmer
* Reorders the `language` values to show recommended algorithms first,
  followed by other values alphabetically

### CODE REVIEW

File: docs/reference/analysis/tokenfilters/stemmer-tokenfilter.asciidoc [hunk 1]
```diff
@@ -4,189 +4,278 @@
 <titleabbrev>Stemmer</titleabbrev>
 ++++
 
-// Adds attribute for the 'minimal_portuguese' stemmer values link.
-// This link contains ~, which is converted to subscript.
-// This attribute prevents that substitution.
-// See https://github.com/asciidoctor/asciidoctor/wiki/How-to-prevent-URLs-containing-formatting-characters-from-getting-mangled
-:min-pt-stemmer-values-url: http://www.inf.ufrgs.br/~buriol/papers/Orengo_CLEF07.pdf
+Provides algorithmic stemming for several languages, some with additional
```
Comment (jrodewig) [inline docs/reference/analysis/tokenfilters/stemmer-tokenfilter.asciidoc:7]: I can xref algorithmic stemming with https://github.com/elastic/elasticsearch/pull/55156/files#diff-8dbd6670213faa5eccc209176d0dd22bR33 when #55156 is merged.

File: docs/reference/analysis/tokenfilters/stemmer-tokenfilter.asciidoc [hunk 2]
```diff
@@ -4,189 +4,278 @@
...
+====
+Valid values are sorted by language. Defaults to
+http://snowball.tartarus.org/algorithms/porter/stemmer.html[*`english`*].
+Recommended algorithms are *bolded*.
 
-[horizontal]
 Arabic::
-
-http://lucene.apache.org/core/4_9_0/analyzers-common/org/apache/lucene/analysis/ar/ArabicStemmer.html[*`arabic`*]
+{lucene-analysis-docs}/ar/ArabicStemmer.html[*`arabic`*]
```
Comment (romseygeek) [inline docs/reference/analysis/tokenfilters/stemmer-tokenfilter.asciidoc:119]: 👍

**Review by romseygeek — APPROVED** [review]
LGTM, thanks @jrodewig

**Comment by jrodewig** [comment]
@elasticmachine update branch

**Comment by jrodewig** [comment]
### Backport commits
master bb9dbcb
7.x e0a8adb
7.7 9ba83ef
7.6 7ffcb9e

**Pull request merged**